### PR TITLE
Functional tests - Add new test "Customers > Outstanding page"  Help card - View order - View Invoice

### PR DESCRIPTION
**Pull request by khouloudbelguith**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | Add a new test for the "Customers > Outstanding page"  Help card - View order - View Invoice
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | HEADLESS=false TEST_PATH="functional/BO/04*/03*" URL_FO="http://shop.com/" npm run test:specific
| Possible impacts? | no

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/04_customers/03_outstanding/02_viewOrder.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+        if (test.args.columnName === 'id_order') {
+          orderId = orderColumn;
+          await expect(orderColumn).to.be.at.least(test.args.result);
+        } else {
+          orderReference = orderColumn;
+          await expect(orderColumn).to.not.equal(test.args.result);
+        }
+      });
+    });
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/04_customers/03_outstanding/02_viewOrder.js]:
```suggestion
      it('should reset filter and get the last orderID and reference', async function () {
        await testContext.addContextItem(this, 'testIdentifier', 'resetFilterOrder', baseContext);

        await ordersPage.resetFilter(page);

        orderId = await ordersPage.getTextColumn(page, 'id_order', 1);
        await expect(orderColumn).to.be.at.least(1);
        
        orderReference = await ordersPage.getTextColumn(page, 'reference', 1);
        await expect(orderReference).to.not.be.null;
    });
```

File: tests/UI/campaigns/functional/BO/04_customers/03_outstanding/02_viewOrder.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    it('should reset filter and get the last outstanding ID', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'resetFilterOutstanding', baseContext);
+
+      await outstandingPage.resetFilter(page);
+
+      outstandingId = await outstandingPage.getTextColumn(page, 'id_order', 1);
+      await expect(outstandingId).to.be.at.least(1);
+    });
+
+    it('should click on view Order', async function () {
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/04_customers/03_outstanding/02_viewOrder.js]:
```suggestion
    it('should view the Order and check the orderID and the reference', async function () {
```

File: tests/UI/pages/BO/customers/outstanding/index.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  /**
+   * Click on view order
+   * @param page {Page} Browser tab
+   * @param columnName {string} Column name on table
+   * @param row {number} Outstanding row in table
+   * @returns {Promise<void>}
+   */
+  async viewOrder(page, columnName, row) {
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/customers/outstanding/index.js]:
```suggestion
  async viewOrder(page, columnName, row = 1) {
```

File: tests/UI/pages/BO/customers/outstanding/index.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  /**
+   * Click on view invoice
+   * @param page {Page} Browser tab
+   * @param columnName {string} Column name on table
+   * @param row {number} Outstanding row in table
+   * @returns {Promise<string>}
+   */
+  async viewInvoice(page, columnName, row) {
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/customers/outstanding/index.js]:
```suggestion
  async viewInvoice(page, columnName, row = 1) {
```

**Comment by khouloudbelguith** [comment]
Hello @nesrineabdmouleh, I fixed the review.

Thank you!

**Review by boubkerbribri — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @khouloudbelguith

**Pull request merged**